ngraph/src/ngraph/graph_util.cpp

```diff
@@ -305,11 +306,14 @@ std::vector<std::shared_ptr<ngraph::Node>>
                 }
             }
             auto cloned_node = node->copy_with_new_inputs(cloned_args, cloned_dependencies);
-            if (node->get_friendly_name() != node->get_name())
+            // There is a friendly name for this node so copy it
+            cloned_node->set_friendly_name(node->get_friendly_name());
+            //  TODO: workaround for shape inference, delete it after fix
+            if (ngraph::as_type_ptr<ngraph::op::TensorIterator>(cloned_node))
             {
-                // There is a friendly name for this node so copy it
-                cloned_node->set_friendly_name(node->get_friendly_name());
+                cloned_node->validate_and_infer_types();
             }
+            cloned_node->get_rt_info() = node->get_rt_info();
```

Comment:
`get_rt_info()` returns reference so if we change rt info in original function it will change in its copy

Reply:
Good catch! The same situation for specialize function. Fixed both cases.